auto_tuning/proxy/src/proxy-runners/run-all.py

```diff
@@ -6,19 +6,22 @@
 import argparse
 import proxy_bindings as pb
 from progress.bar import Bar
-import json
+import pandas as pd
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import time
+import os
 
 
 parser = argparse.ArgumentParser()
 parser.add_argument('-c', '--cells', default=50000, type=int, help="num cells in a time cluster")
 parser.add_argument('-t', '--timesteps', default=30, type=int, help="num time steps/repeats")
-parser.add_argument('-o', '--output_type', default='c', type=str, help="console(c), json (j)")
+parser.add_argument('--output_type', default='c', type=str, help="console(p), json (j), matplotlib (m)")
+parser.add_argument('-o', '--output_dir', default='.', type=str, help="relative output directory")
 args = parser.parse_args()
 
-if not args.output_type in ['j', 'c']:
-  print(f'error: expected `j` of `c` as output type, given `{args.output_type}`')
-  parser.print_help()
-  sys.exit(-1)
+if not args.output_type in ['j', 'c', 'p']:
+    print(f'notice: output only to terminal')
```

Comment:
You, can use the `choices` option: https://docs.python.org/3/library/argparse.html#choices to make the help redundant.
Also I don't completly understand your logic. You give options `p, j, m` in the help, but then ask for `j, c, p` later.

Reply:
Completely forgot about `choices`. Thanks for reminding!

> You give options p, j, m in the help, but then ask for j, c, p later

sorry, a bunch of typos

Reply:
replaced with 

```
plotting_options = ['csv, json', 'matplotlib']
...
parser.add_argument('--output_type', choices=plotting_options, default='csv', help='format to save')
```
done